kernel/src/process.rs

```diff
@@ -1923,10 +1921,6 @@ impl<C: 'static + Chip> Process<'_, C> {
         let initial_kernel_memory_size =
             grant_ptrs_offset + Self::CALLBACKS_OFFSET + Self::PROCESS_STRUCT_OFFSET;
 
-        if min_app_ram_size < Self::INITIAL_APP_MEMORY_SIZE {
-            min_app_ram_size = Self::INITIAL_APP_MEMORY_SIZE;
-        }
```

Comment:
I think we still need this check (or a version of it). If the TBF sets the requested RAM to 0 (or less than arch.syscall says is the minimum), then we might put the stack pointer inside of the kernel reserved memory.

Reply:
I added a check to verify the initial process memory region and initial kernel memory region do not overlap. I decided to make it an error rather than quietly increasing the process memory region's size for two reasons:

A. This likely indicates something is wrong with the `min_ram_size` calculation.
B. It is very unlikely the process will be happy with only 32 bytes of usable memory, and it would not be able to increase its break.